src/pages/Lending/hooks/useRepaymentLockData.tsx

```diff
@@ -0,0 +1,92 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { getThorchainLendingPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/utils'
+
+type UseLendingPositionDataProps = {
+  accountId?: AccountId
+  assetId?: AssetId
+}
+
+// non-exhaustive, we only use this to get the current blockheight
+type ThorchainBlock = {
+  header: {
+    height: number
+  }
+}
+
+export const useRepaymentLockData = ({ accountId, assetId }: UseLendingPositionDataProps) => {
+  const repaymentLockQueryKey: [
+    string,
+    { accountId: AccountId | undefined; assetId: AssetId | undefined },
+  ] = useMemo(() => ['thorchainLendingRepaymentLock', { accountId, assetId }], [accountId, assetId])
+
+  const repaymentLockData = useQuery({
+    staleTime: Infinity,
+    queryKey: repaymentLockQueryKey,
+    queryFn: async () => {
+      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+
+      const maybePosition = await (async () => {
+        if (!(accountId && assetId)) return null
```

Comment:
can get rid of this and do all three calls in parallel. you are already checking for position in the selector and using `!` for block height currently